met-api/src/met_api/models/widget.py

```diff
@@ -0,0 +1,70 @@
+"""Widget model class widget.
+
+Manages the widget
+"""
+from __future__ import annotations
+from datetime import datetime
+
+from sqlalchemy.sql.schema import ForeignKey
+
+
+from .widget_item import WidgetItem
+
+from .db import db
+
+
+class Widget(db.Model):  # pylint: disable=too-few-public-methods
+    """Definition of the Widget entity."""
+
+    __tablename__ = 'widget'
+    __table_args__ = (
+        db.UniqueConstraint('widget_type_id', 'engagement_id', name='unique_widget_type'),
+    )
+
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    widget_type_id = db.Column(db.Integer, ForeignKey('widget_type.id', ondelete='CASCADE'))
+    engagement_id = db.Column(db.Integer, ForeignKey('engagement.id', ondelete='CASCADE'))
+    created_date = db.Column(db.DateTime, default=datetime.utcnow, nullable=False)
+    updated_date = db.Column(db.DateTime, onupdate=datetime.utcnow, nullable=False)
+    created_by = db.Column(db.String(50), nullable=False)
+    updated_by = db.Column(db.String(50), nullable=False)
+    items = db.relationship('WidgetItem', backref='widget', cascade='all, delete')
+
+    @classmethod
+    def get_widgets_by_engagement_id(cls, engagement_id):
+        """Get widgets by engagement_id."""
+        return db.session.query(Widget)\
+            .join(WidgetItem, Widget.id == WidgetItem.widget_id, isouter=True)\
+            .filter(Widget.engagement_id == engagement_id)\
+            .order_by(Widget.id.desc())\
+            .all()
+
```

Comment:
plz add a return type 
Also if we are just getting widgets by this query , why join the widget item?
Add a comment to explain the logic , will be helpful

Reply:
Widget and widget items are dependent. When getting them we always want them together. They are separate tables to make creating and updating easier